hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -454,13 +454,13 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		ObservedGeneration: nodePool.Generation,
 	})
 
-	// Validate arch platform support
-	if (nodePool.Spec.Arch != "") && (nodePool.Spec.Platform.Type != hyperv1.AWSPlatform) {
+	// Validate modifying CPU arch support for platform
```

Comment:
@bryan-cox can we add CEL validation for this https://kubernetes.io/docs/reference/using-api/cel/?

Reply:
Opened [HOSTEDCP-1262](https://issues.redhat.com/browse/HOSTEDCP-1262) to capture this request.